### DISCUSSION TITLE
How to migrate a container

### DISCUSSION THREAD

**Discussion by markc** [lxdware/lxd-dashboard#37] [Q&A]
I'm using v3.3 on an Ubuntu 22.04 Proxmox VM in a 3 node cluster, and for the life of me I cannot find a web interface option to migrate a CT to another node. Googling "lxdware migrate container" did not help. Is it possible? A hint as to how to do this via lxdware please?

Update: well, that was interesting (to me) as a LXD novice. I tried to migrate an instance using the CLI and I could not even create a stateful snapshot, so it seems that any kind of LIVE migration will never work unless the instances are very carefully curated and, for instance, do not use systemd, which leaves out stock Ubuntu as a guest.

FWIW I am used to using Proxmox where live VM migration works well. However, their CT migration requires stopping the CT instance first, so I guess they have already bumped into this issue and don't even offer live CT migration at all. Now I understand why.

**Comment by matthewalanpenning** [comment]
I had rearragned some code in the 3.2.0 release and thought I should test migrations on VMs just to verify things are working on the dashboard (Initial testing was only done on containers). In the process I discovered that the selection drop-drop down was not providing choices to select a cluster host, so I pushed that fix out in the 3.4.0 release. Thanks for the question, it helped me find a bug in the code. 

Offline migration between cluster members works for both containers and virtual machines using the dashboard. Once the instance is shut down an option is provided in the hamburger menu of the instance. This will alllow you to select a cluster member and begin the migration.

As far as live migrations, it has been hit-or-miss for me using the CLI. I had some trouble in the past with setting up the require CRIU and getting the migrations to work on older versions of LXD. I postponed adding live migration option into the dashboard until I had a really good understanding of how LXD performs the operation.

I used the following posts as a guide:
https://ubuntu.com/blog/live-migration-in-lxd
https://discuss.linuxcontainers.org/t/does-lxd-support-live-migration/9070

When I get some time here soon I will revisit the live migration option

**Reply by markc** [reply]
FWIW, don't waste your time with live migration until the LXD folks have it working, including patches to the upstream CRIU project.

I finally got my little cluster working, but it took a couple of full days with a lot of useless tutorials and YouTube videos. Nothing showed me a straightforward, clear-cut step-by-step guide on how to set up a simple 3 node cluster, and then how to move instances between nodes. Now that I can see how it works, this alias and bash function save me a lot of time...
```
alias lx='lxc list'
lxm () 
{ 
    [[ -z $3 || $1 =~ -h ]] && echo "Usage: lxm name1 name2 node" && return 1;
    lxc stop $1;
    lxc move $1 $2 --target $3;
    lxc start $2
}
```
lxm (for "lxd move") is just wonderful, and I use "lxc list" so often that "lx" makes a lot of sense (to me).

Now, I have a semi-suggestion. When I go to lxd-dashbord I find I would really like to see what is basically the output of lxc list without having to go to a node first and then select either "Containers" or "Virtual Machines". I can appreciate how you have split it up so that if there were 100s of CT/VMs then it could be better, but I'd be looking at the least number of clicks to get to the important CT/VM info.
